test/valid-data/string-template-expression-literals/main.ts

```diff
@@ -2,9 +2,11 @@ type OK = "ok";
 type Result = OK | "fail" | `abort`;
 type PrivateResultId = `__${Result}_id`;
 type OK_ID = `id_${OK}`;
+type Num = `${number}`;
```

Comment:
We also should get string for booleans inside, no?

Reply:
Agreed, I'll add a test